wordpress-mobile/WordPress-Android
# Stats revamp v2 views and visitors card

This PR adds new Views and Visitors card in Insights tab on Stats screen.

Fixes #16203 

<img width=320 src="https://user-images.githubusercontent.com/990349/161727418-7acb9716-714c-4b03-9096-cc31ecf0049c.png" />


To test:

Setup:

- Go to App Settings (Tap on Avatar at the top right hand corner on My Site -> Find App Settings)
- Select Debug Settings
- Find StatsRevampV2FeatureConfig under Features in development enable it and restart app

Test 1:
- Launch app
- Go to stats either using quick links or menu 
- Notice a new **Views & Visitors** card is shown in Insights tab as in the image above (scroll down if necessary)

Test 2:
- Launch Stats management (While on Stats in Insights tab, Click on ⚙️ icon at the top right hand corner)
- Notice there's **Views & Visitors** item under General 
- Tap on **Views & Visitors** item to deselect it and hit SAVE
- Return to Stats and observe the **Views & Visitors** Card is no longer shown
- You may attempt selecting and deselecting a few times to make sure it adds or removes accordingly

Known Issues:
- The card is added to the end of the list as per existing implementation unless app is a fresh install in which case this card will appear at the top.  Also, when Stats Management is used to add or remove the card gets added to the end of the list
- Chart will be updated in a separate PR as part of next issue
- Styles will be updated as per design in a separate PR

TODO:
- [x] Need to add/update tests
- [x] [Corresponding FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2340) needs to be merged first before this


## Regression Notes
1. Potential unintended areas of impact
Stats Insights 

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual and unit tests

3. What automated tests I added (or what prevented me from doing so)
Unit tests

PR submission checklist:
